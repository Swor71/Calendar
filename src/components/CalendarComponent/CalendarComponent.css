.calendar-component__wrapper {
  margin: 0 auto;
  min-height: 400px;
  width: 400px;
  border: 1px solid #dee0e4;
  border-radius: 4px;
  background: #fff;
  padding: 20px;
  position: relative;
}

.calendar-component__wrapper::before {
  content: '';
  position: absolute;
  height: 16px;
  width: 16px;
  top: -10px;
  left: 26px;
  background: #fff;
  border-right: 1px solid #dee0e4;
  border-top: 1px solid #dee0e4;
  transform: rotate(-45deg);
}
