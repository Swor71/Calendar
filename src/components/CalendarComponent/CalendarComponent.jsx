import React, { Component } from 'react';
import CalendarNav from './CalendarNav/CalendarNav';
import CalendarBody from './CalendarBody/CalendarBody';
import CalendarFooter from './CalendarFooter/CalendarFooter';
import PropTypes from 'prop-types';

import './CalendarComponent.css';

class CalendarComponent extends Component {
  static propTypes = {
    availableDates: PropTypes.arrayOf(PropTypes.object),
    calendarData: PropTypes.object,
    onHandleSelectDate: PropTypes.func,
    onHandleMonthChange: PropTypes.func,
  }

  static defaultProps = {
    availableDates: [{}],
    calendarData: {
      currentDate: '',
      activeInput: '',
      daysInMonth: [],
      emptyBlocks: [],
    },
  }

  render() {
    const { calendarData, availableDates, onHandleSelectDate, onHandleMonthChange } = this.props;
    const { currentDate, activeInput } = calendarData;
    const selectedInput = `calendar-component__wrapper ${(activeInput === 'checkOutDate') ? 'checkOutDate' : 'checkInDate'}`;

    return (
      <div className={selectedInput}>
        <CalendarNav
          currentDate={currentDate}
          onHandleMonthChange={onHandleMonthChange}
        />
        <CalendarBody
          calendarData={calendarData}
          availableDates={availableDates}
          onHandleSelectDate={onHandleSelectDate}
        />
        <CalendarFooter />
      </div>
    );
  }
}

export default CalendarComponent;
