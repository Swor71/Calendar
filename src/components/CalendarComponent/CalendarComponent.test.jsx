import React from 'react';
import CalendarComponent from './CalendarComponent';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('CalendarComponent', () => {
  it('renders the component', () => {
    const component = shallow(<CalendarComponent currentDate={'January 2019'} />);
    expect(component.length).toEqual(1);
  });

  it('renders correctly', () => {
    const component = renderer
      .create(<CalendarComponent currentDate={'January 2019'} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
