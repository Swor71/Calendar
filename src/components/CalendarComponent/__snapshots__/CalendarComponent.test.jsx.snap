// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarComponent renders correctly 1`] = `
<div
  className="calendar-component__wrapper checkInDate"
>
  <div
    className="calendar-nav__wrapper"
  >
    <button
      className="calendar-nav__btn--arrow-left"
      onClick={[Function]}
    >
      ⟵
    </button>
    <span
      className="calendar-nav__date"
    >
      
    </span>
    <button
      className="calendar-nav__btn--arrow-right"
      onClick={[Function]}
    >
      ⟶
    </button>
  </div>
  <div
    className="calendar-body__wrapper"
  >
    <div
      className="weekday"
    >
      Su
    </div>
    <div
      className="weekday"
    >
      Mo
    </div>
    <div
      className="weekday"
    >
      Tu
    </div>
    <div
      className="weekday"
    >
      We
    </div>
    <div
      className="weekday"
    >
      Th
    </div>
    <div
      className="weekday"
    >
      Fr
    </div>
    <div
      className="weekday"
    >
      Sa
    </div>
  </div>
  <div
    className="calendar-footer__wrapper"
  >
    <span>
      Minimum stay varies
    </span>
    <span>
      Updated 23 days ago
    </span>
  </div>
</div>
`;
